### PR TITLE
Don't allow rename in inline widget if that identifier have reference…

### PR DESCRIPTION
**Pull request by navch**
This PR is in continuation of https://github.com/adobe/brackets/pull/13965.
This PR will help in deciding weather rename can be done or not inside inline widget.

Please review @nethip @swmitra @petetnt @boopeshmahendran

### CODE REVIEW

File: src/extensions/default/JavaScriptRefactoring/RenameIdentifier.js [hunk 1]
```diff
@@ -100,11 +100,27 @@ define(function (require, exports, module) {
          * If yes then select all references
          */
         function handleFindRefs (refsResp) {
+            var inlineWidget = EditorManager.getFocusedInlineWidget(),
+                editor = EditorManager.getActiveEditor();
+            //In case of inline widget if some references are outside widget's text range then don't allow for rename
+            if (inlineWidget) {
+                var isInTextRange = true;
+                refsResp.references.refs.forEach(function(item) {
```
Comment (petetnt) [inline src/extensions/default/JavaScriptRefactoring/RenameIdentifier.js]: `refsResp.references.refs` might throw. In #L119 it's handled by checking for `if (resResp && ...)`.

I think easiest thing would be null checking it in the start of the function and either returning immediately if it's null and/or saving it to a variable if it is.

```js
function handleFindRefs (refsResp) {
  if (refsResp && refsResp.references && refsResp.references.refs) {
     return;
  }

  var refs = refsResp.references.refs;
  var type = refsResp.references.type;

  ...
}```

**Review by petetnt — CHANGES_REQUESTED** [review]

**Comment by petetnt** [comment]
Cool stuff! Left a small comment.

**Comment by navch** [comment]
Thanks @petetnt  for reviewing this. Sorry for late response. I addressed review comment. Please have a look.

**Review by petetnt — APPROVED** [review]
Codewise LGTM

File: src/extensions/default/JavaScriptRefactoring/RenameIdentifier.js [hunk 2]
```diff
@@ -100,15 +100,36 @@ define(function (require, exports, module) {
...
+            }
+
+            var inlineWidget = EditorManager.getFocusedInlineWidget(),
+                editor = EditorManager.getActiveEditor(),
+                refs = refsResp.references.refs,
+                type = refsResp.references.type;
+
+            //In case of inline widget if some references are outside widget's text range then don't allow for rename
+            if (inlineWidget) {
+                var isInTextRange = true;
```
Comment (petetnt) [inline src/extensions/default/JavaScriptRefactoring/RenameIdentifier.js]: Can we use `Array.prototype.find` on all platforms? Would simplify this to
```js
var isInTextRange  = !refs.find(function(item) {
  return (item.start.line < inlineWidget._startLine || item.end.line > inlineWidget._endLine);
});
```

Comment (navch) [reply src/extensions/default/JavaScriptRefactoring/RenameIdentifier.js]: 👍 Done :)

**Pull request merged**